src/main/resources/META-INF/services/org.apache.lucene.codecs.PostingsFormat

```diff
@@ -0,0 +1,24 @@
+# Copyright (c) 2017 Rocana
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This is an SPI (service provider interface) config file so Lucene can look up
+# our custom postings format at runtime using just the fully qualified class name
+# here.
+#
+# If you have access to the Rocana JIRA, see ROCANA-8229.
+
+com.rocana.lucene.codec.v1.RocanaLucene50PostingsFormat
```

Comment:
I originally had it in the file name. @joey suggested moving it to the package name, and then in a PR review with you we talked about putting it back in the file name, so it would be in both.

But if I remove the version from the package name now then I've violated something @joey and I talked about and agreed to.

As to why we agreed to it, I'm sure Joey could defend it better than I could. I think the idea was we'd need to support two codecs in the future, both simultaneously, so we need the version in the package. I see that going two routes:

1. When we need a v2 codec we aren't also upgrading Lucene, in which case we're either our v2 codec is entirely custom (in which case there's no conflict and the separate package works out well), or it's based on some other Lucene classes we're not using yet (this also works out well for the v2 package), or we further customize our existing forked code, to the point where it's changing the on-disk representation enough that we need to increment the codec version to v2. That last one is the hairiest, but in that case I think we'd have to copy/paste the existing v1 code to the v2 package and then tweak it there. The reason I think we need this is because we'd need both the v1 and v2 codec at the same time.
2. We upgrade Lucene, which will be really messy, but in this case our v2 codec might be based off Lucene 7.0, for example, and maybe Lucene 7.0 has a different on disk format. In this case we can only have one version of Lucene on the classpath at once, and it would have to be the newer version. So we're relying on new Lucene still having the Lucene54 codec code. If so, then our v1 package can stay unmodified and our v2 package can contain newer versions of those files. Or, maybe some of those files aren't specific to a codec. In that case maybe we move them from v1 to a common package, or upgrade our forks to the latest Lucene version and then move them to a common package since they're shared between codecs.

I'm not sure how it's going to work out, but if we need a v2 codec in the future, we'll know more then. (Of course, you could make that same argument that we shouldn't put the version in the package until we know we need a v2 codec, and I can't argue with that).